packages/server/modules/serverinvites/domain/types.ts

```diff
@@ -1,22 +1,48 @@
-import { ResourceTargets } from '@/modules/serverinvites/helpers/inviteHelper'
+import {
+  ProjectInviteResourceType,
+  ServerInviteResourceType
+} from '@/modules/serverinvites/domain/constants'
 import { Nullable } from '@/modules/shared/helpers/typeHelper'
-import { SetNonNullable } from 'type-fest'
+import { ServerRoles, StreamRoles } from '@speckle/shared'
 
-export type ServerInviteRecord = {
-  id: string
-  target: string
-  inviterId: string
-  createdAt: Date
-  used: boolean
-  message: Nullable<string>
-  resourceTarget: typeof ResourceTargets.Streams | null
-  resourceId: Nullable<string>
-  role: Nullable<string>
-  token: string
-  serverRole: Nullable<string>
+export interface InviteResourceTargetTypeMap {
```

Comment:
Might seem unnecessary, but this allows for dynamic interface augmentation from the workspaces module